api/src/neo4j/queries/interactionPartners.js

```diff
@@ -49,7 +50,125 @@ WITH apoc.map.mergeList(apoc.coll.flatten(
 )) as reaction, component
 RETURN { component: component, reactions: COLLECT(reaction)}
 `;
-  return querySingleResult(statement);
+  const result = await querySingleResult(statement);
+  let links = [];
+  let nodes = [];
+  let unique = new Set();
+
+  const addLink = (s, t) => {
+    const link = `${s}-${t}`;
+    const inverseLink = `${t}-${s}`;
+    if (!unique.has(link) && !unique.has(inverseLink)) {
+      links.push({ s, t });
+      unique.add(link);
+    }
+  };
+
+  result.reactions.forEach(reaction => {
+    const { genes, metabolites } = reaction;
+
+    // loop through metabolites, add them to nodes
+    // and add links to the main node
+    metabolites.forEach(metabolite => {
+      if (!unique.has(metabolite.id)) {
+        nodes.push({
+          g: 'm',
+          id: metabolite.id,
+          n: metabolite.name || metabolite.name,
+        });
+        unique.add(metabolite.id);
+
+        if (id !== metabolite.id) {
+          addLink(id, metabolite.id);
+        }
+      }
+    });
+
+    // loop through genes, add them to nodes
+    // and add links to the main node
+    genes.forEach(gene => {
+      if (!unique.has(gene.id)) {
+        nodes.push({ g: 'e', id: gene.id, n: gene.name || gene.id });
+        unique.add(gene.id);
+
+        if (id !== gene.id) {
+          addLink(id, gene.id);
+        }
+      }
+
+      // loop through metabolites for each gene
+      // and add links to the gene
+      metabolites.forEach(metabolite => addLink(gene.id, metabolite.id));
+    });
+  });
+  const network = await populateWithLayout({
+    nodes,
+    links,
+    dim: 2,
+    mainNodeID: id,
+    reCenter: true,
+  });
+  return { result, network };
+};
+
+const getInteractionPartnersExpansion = async ({
+  id,
+  model,
+  version,
+  expanded,
+}) => {
+  const { network, result } = await getInteractionPartners({
+    id,
+    model,
+    version,
+  });
+  let unique = new Set();
+  let expandedNodes = {};
+  // loop through all expanded nodes and add them to the network
+  for (const nodeId of expanded) {
+    // eslint-disable-next-line no-await-in-loop
+    const expandedNetwork = await getInteractionPartners({
```

Comment:
Instead of looping through this and adding this eslint disable, maybe we could use `await Promise.all` instead.